chassis/core/src/main/java/com/griddynamics/jagger/master/MasterToJaasCoordinator.java

```diff
@@ -185,9 +188,7 @@ public boolean isRegistered() {
         
         private volatile RequestEntity<TestEnvironmentEntity> requestEntity;
 
-        private TestEnvironmentStatus lastStatus = TestEnvironmentStatus.PENDING;
-
-        private volatile boolean runningStatusSent = false;
+        private final Object statusLock = new Object();
```

Comment:
For this case synchronized is better.
Performance of synchronized and ReentrantLock is the same, but ReentrantLock needs a try-catch-finally block which makes code dirtier, and synchronized doesn't need it.